packages/cli/src/bin/pipcook.ts

```diff
@@ -45,6 +52,39 @@ export const cacheClean = async (opts: CacheCleanOptions): Promise<void> => {
   await Promise.all(futures);
 };
 
+export const createScriptRepo = async (scriptType: string, name: string): Promise<void> => {
+  try {
+    const destDir = join(process.cwd(), name);
+    const pkgFile = join(destDir, 'package.json');
+    if (await fs.pathExists(destDir)) {
+      throw new TypeError(`${name} already exists`);
+    }
+    const [ type, branch = 'main' ] = scriptType.split('@');
+    const template = `${templateMap[type]}#${branch}`;
+    if (!template) {
+      throw new TypeError(`no template found for ${scriptType}, it should be one of 'datasource', 'dataflow', 'model'`);
```

Comment:
It's good to generate the string by `templateMap`.

Reply:
Can you describe it more?